### ISSUE TITLE
Flutter 2.5.0 embedding V2 problem

### ISSUE THREAD

**Issue by tokyokamera** [kmcgill88/admob_flutter#286]
I have a problem as below. How can I fix this?

**A red warning when run "flutter pub get"**

```
The plugins `admob_flutter` use a deprecated version of the
Android embedding.
To avoid unexpected runtime failures, or future build failures, try to see if
these plugins support the Android V2 embedding. Otherwise, consider removing
them since a future release of Flutter will remove these deprecated APIs.
If you are plugin author, take a look at the docs for migrating the plugin to
the V2 embedding: https://flutter.dev/go/android-plugin-migration.
```

**This is my flutter version:**

```
[Flutter (Channel stable, 2.5.0, on macOS 11.2.3 20D91 darwin-x64, locale ja-JP)
    ｢ Flutter version 2.5.0 at /Applications/flutter
    ｢ Upstream repository https://github.com/flutter/flutter.git
    ｢ Framework revision 4cc385b4b8 (2 days ago), 2021-09-07 23:01:49 -0700
    ｢ Engine revision f0826da7ef
    ｢ Dart version 2.14.0

[Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    ｢ Android SDK at /Users/tokyokamera/Library/Android/sdk
    ｢ Platform android-30, build-tools 30.0.2
    ｢ Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    ｢ Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)
    ｢ All Android licenses accepted.

[Xcode - develop for iOS and macOS
    ｢ Xcode at /Applications/Xcode.app/Contents/Developer
    ｢ Xcode 12.5, Build version 12E262
    ｢ CocoaPods version 1.10.1

[Chrome - develop for the web
    ｢ Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[Android Studio (version 4.0)
    ｢ Android Studio at /Applications/Android Studio.app/Contents
    ｢ Flutter plugin version 48.1.2
    ｢ Dart plugin version 193.7361
    ｢ Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)

[Connected device (1 available)
    ｢ Chrome (web) ｢ chrome ｢ web-javascript ｢ Google Chrome 93.0.4577.63
```

**Comment by bahcem** [comment]
is there any solution for this ?

**Comment by tokyokamera** [comment]
I found a solution to this problem.

1. Download the code for the plugin (https://github.com/kmcgill88/admob_flutter/archive/refs/heads/master.zip) and place it at the "same" level as your flutter project directory.
2. Now open your pubsec.yaml and write code as follows
//  admob_flutter: ^1.0.1
  admob_flutter:
      path: ./admob_flutter/
3. Run "pub get" , it will point to the local file.

**Comment by tokyokamera** [comment]
So sorry, It's not working on MacOS.

**Comment by creativeandresearch** [comment]
is there any solution for this?

**Comment by kmcgill88** [comment]
Unfortunately I haven't had time to work on this I'm happy to merge any PRs with a fix!

**Comment by kyleorin** [comment]
any update on this?

**Comment by cihadturhan** [comment]
Is there any solution to this? Or, what is the lastest working version of it?

**Comment by cihadturhan** [comment]
I've tried forking and upgrading for flutter v2.5+ but I guess we need someone who has better knowledge on old and new plugin development.

Just posting steps if anyone intersted :)
https://flutter.dev/docs/development/packages-and-plugins/plugin-api-migration#upgrade-steps

**Comment by Gintasz** [comment]
Hey, I'm paying $150 if anybody can fix this issue in 1 week (the task is to have the library working in the latest flutter version). With this ad library, we used banner refresh capability between tabs, that increased ad revenue by 2x as the client says, but it's not working in another official AdMob library.

@tokyokamera @bahcem @creativeandresearch @kmcgill88 @kyleorin @cihadturhan

**Comment by kmcgill88** [comment]
Resolved in https://github.com/kmcgill88/admob_flutter/pull/289. published in 2.0.0

**Comment by kmcgill88** [comment]
> Hey, I'm paying $150 if anybody can fix this issue in 1 week (the task is to have the library working in the latest flutter version). With this ad library, we used banner refresh capability between tabs, that increased ad revenue by 2x as the client says, but it's not working in another official AdMob library.
> 
> 
> 
> @tokyokamera @bahcem @creativeandresearch @kmcgill88 @kyleorin @cihadturhan 

What do you think @Gintasz?
